src/filename_inspector.py

```diff
@@ -0,0 +1,106 @@
+import fnmatch
+import glob
+import os
+import json
+import csv
+
+
+def get_input(name: str) -> str:
+    value = os.getenv(f'INPUT_{name.upper()}')
+    return value
+
+
+def set_output(name: str, value: str):
+    output_file = os.getenv('GITHUB_OUTPUT', None)
+    if output_file:
+        with open(output_file, 'a') as f:
+            f.write(f'{name}={value}\n')
+    else:
+        raise EnvironmentError('GITHUB_OUTPUT environment variable is not set.')
+
+
+def set_failed(message: str):
+    print(f'::error::{message}')
```

Comment:
```suggestion
    logger.warning(message)
```

I've intentionally opted for `warning`; we could even settle on `info` as this is the expected outcome, not an error in the programming.

Reply:
Applied in commit bd737a91af6446764f3ee9eadc0edd11903f1354.

Reply:
Returned in commit 994f31093fbfb9e47abced9f7b8d1ad2d4235dfc as this is a common concept to report failure.

Reply:
Final change in commit 57c06c95c7dad8ac405418f94525273fa0bfb417.